libsupport/src/ProgressTracer.cpp

```diff
@@ -152,6 +152,21 @@ katana::ProgressTracer::SetActiveSpan(
   return ProgressScope(std::move(span));
 }
 
+void
```

Comment:
On https://github.com/KatanaGraph/katana/pull/446/files#diff-7fc4d678da093aebb69172abdffa2f315e33051b081466050d8b4497f15dfbd2R131 above:

```
    if (active_span_ != nullptr &&
        (active_span_->ScopeClosed() || active_span_->IsFinished())) {
      active_span_->Finish();
    }
```

it seems weird to me that we finish spans that are already finished.

Reply:
That code path is for handling someone prematurely finishing a span on the active path with an explicit call to span.Finish()
The span is not finished twice since there is a check, but it goes through the logic of setting the active_span to being the parent of the prematurely closed currently active_span

Reply:
Ideally there would a name or state associated with "finished but not really completely finished"; if there is no good way to make that evident in the code, can you at least add a comment why something may be considered "finished" but still needs "finishing"?